ibm/service/kubernetes/resource_ibm_container_ingress_instance.go

```diff
@@ -0,0 +1,252 @@
+// Copyright IBM Corp. 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package kubernetes
+
+import (
+	"fmt"
+	"time"
+
+	v2 "github.com/IBM-Cloud/bluemix-go/api/container/containerv2"
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceIBMContainerIngressInstance() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMContainerIngressInstanceCreate,
+		Read:     resourceIBMContainerIngressInstanceRead,
+		Update:   resourceIBMContainerIngressInstanceUpdate,
+		Delete:   resourceIBMContainerIngressInstanceDelete,
+		Exists:   resourceIBMContainerIngressInstanceExists,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"instance_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Instance CRN id",
+			},
+			"cluster": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Cluster ID",
+				ValidateFunc: validate.InvokeDataSourceValidator(
+					"ibm_container_ingress_instance",
+					"cluster"),
+			},
+			"instance_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Instance registration name",
+			},
+			"secret_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
```

Comment:
Why is default "" ? If user doesn't provide secret_group_id but API returns back as part of GET then we should make it computed as well.

Reply:
The user can provide the secret_group_id but it is an optional parameter

Reply:
Fixed to be marked as a computed field